packages/snaps-controllers/src/interface/utils.ts

```diff
@@ -55,13 +100,76 @@ export function assertNameIsUnique(state: InterfaceState, name: string) {
   );
 }
 
+/**
+ * Get a default asset for a given address.
+ *
+ * @param addresses - The account addresses.
+ * @param chainIds - The chain IDs to filter the assets.
+ * @param elementDataGetters - Data getters for the element.
+ * @param elementDataGetters.getAccountByAddress - A function to get an account by its address.
+ * @param elementDataGetters.getAssetsState - A function to get the MultichainAssetController state.
+ *
+ * @returns The default asset for the account or undefined if not found.
+ */
+export function getDefaultAsset(
+  addresses: CaipAccountId[],
+  chainIds: CaipChainId[] | undefined,
+  { getAccountByAddress, getAssetsState }: ElementDataGetters,
+) {
+  const { assetsMetadata, accountsAssets } = getAssetsState();
+
+  const parsedAccounts = addresses.map((address) =>
+    parseCaipAccountId(address),
+  );
+
+  const accountChainIds = parsedAccounts.map(({ chainId }) => chainId);
+
+  const filteredChainIds =
+    chainIds && chainIds.length > 0
+      ? accountChainIds.filter((accountChainId) =>
+          chainIds.includes(accountChainId),
+        )
+      : accountChainIds;
+
+  const accountId = getAccountByAddress(addresses[0])?.id;
+
+  // We should never fail on this assertion as the address is already validated.
+  assert(accountId, `Account not found for address: ${addresses[0]}.`);
+
+  const accountAssets = accountsAssets[accountId];
+
+  if (accountAssets.length === 0) {
+    return undefined;
+  }
+
+  const nativeAsset = accountAssets.find((asset) => {
+    const { chainId, assetNamespace } = parseCaipAssetType(asset);
+
+    return filteredChainIds.includes(chainId) && assetNamespace === 'slip44';
+  });
+
+  if (!nativeAsset) {
+    return {
+      address: accountAssets[0],
+      ...assetsMetadata[accountAssets[0]],
+    };
+  }
+
+  return {
+    address: nativeAsset,
+    ...assetsMetadata[nativeAsset],
+  };
+}
```

Comment:
Perhaps flip these for readability, if native asset exists return it, fallback to asset 0